cmd/collector/app/server/zipkin.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/cors"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/collector/app/handler"
+	"github.com/jaegertracing/jaeger/cmd/collector/app/zipkin"
+	"github.com/jaegertracing/jaeger/pkg/healthcheck"
+)
+
+// ZipkinServerParams to construct a new Jaeger Collector Zipkin Server
+type ZipkinServerParams struct {
+	Port            int
+	Handler         handler.ZipkinSpansHandler
+	RecoveryHandler func(http.Handler) http.Handler
+	AllowedOrigins  string
+	AllowedHeaders  string
+	HealthCheck     *healthcheck.HealthCheck
+	Logger          *zap.Logger
+}
+
+// StartZipkinServer based on the given parameters
+func StartZipkinServer(params *ZipkinServerParams) (*http.Server, error) {
+	if params.Port == 0 {
```

Comment:
Why is there this check? We normally do not do these checks unless there is a convention in this API that it will no start the server if the port is `0`.

Reply:
That's the Zipkin server: we'll not start it if the port hasn't been specified.

Reply:
I see this is conventions we use for zipkin port flag.